greedy_approach/prim.c

```diff
@@ -0,0 +1,209 @@
+/**
+ * @file
+ * @author [Timothy Maloney](https://github.com/sl1mb0)
+ * @brief [Prim's algorithm](https://en.wikipedia.org/wiki/Prim%27s_algorithm) implementation
+ * implementation of prim's algorithm in C to find the MST of a weighted, connected graph.
+ * @details Prim's algorithm uses a greedy approach to generate the MST of a weighted connected graph.
+ * The algorithm begins at an arbitrary vertex v, and selects a next vertex u, 
+ * where v and u are connected by a weighted edge whose weight is the minimum of all edges connected to v. 
+ * @references Page 319 "Introduction to the Design and Analysis of Algorithms" - Anany Levitin
+ *
+ * To test - run './prim -test'
+ * prim() will find the MST of the following adj. matrix:
+ *	  
+ *	  0  1  2  3
+ *        1  0  4  6
+ *        2  4  0  5
+ *        3  6  5  0
+ * 
+ * The minimum spanning tree for the above weighted connected graph is given by the following adj matrix:
+ *	   
+ *	  0  1  3  2
+ *	  1  0  0  0
+ *	  3  0  0  0
+ *	  2  0  0  0
+ *
+ *
+ * The following link provides a visual representation of graphs that can be used to test / verify the algorithm for different adj
+ * matrices and their weighted, connected graphs.
+ * (https://visualgo.net/en/mst)
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+
+#define MAX 20
+#define INF 999
+
+int G[MAX][MAX];    // weighted, connected graph G
+int MST[MAX][MAX];  // adj matrix to hold minimum spanning tree of G
+int V;              // number of vertices in V in G
+
+/**
+ * @brief Function minimum
```

Comment:
Please add proper documentation here.

Reply:
What needs to be added? I removed the global variables so I'm not sure what you mean.

Reply:
A brief description of the function is what the functions are for, how does it work, etc. but in short terms. For the detailed description, use `@details`. 🙂